### DISCUSSION TITLE
How to replace one plugin (source) with another

### DISCUSSION THREAD

**Discussion by PerMalmberg** [LazyVim/LazyVim#3238] [Q&A] [answered]
Hello,

What is the correct approach to replace one version of a plugin with another? I've got a fork for nvim-dap-go with a fix waiting to get merged here: https://github.com/PerMalmberg/nvim-dap-go. Until that happens, can I replace the original one being loaded via Lazy with my fork or do I have to replace the entire lang.go config?

I'm really new to both Nvim and Lazy so please ELI5, thank you.

**Comment by dpetka2001** [comment]
I do the same for `nvim-colorizer`, see [here](https://github.com/dpetka2001/dotfiles/blob/edcd80e8c0600d674c662b53df7bf5b9efa39e0e/dot_config/nvim/lua/plugins/tools.lua#L62-L63).

**Reply by PerMalmberg** [reply]
Right, but you have a custom config file for that plugin. I've only enabled lang.go via LazyExtras. I'm thinking there might be a way to override the url and branch: https://github.com/LazyVim/LazyVim/blob/ad22adab7d446018875de2dc96168295cad9298e/lua/lazyvim/plugins/extras/lang/go.lua#L141

Maybe there isn't, but considering the apparent configurability of Lazy, this doesn't seem an impossibility. If there isn't, I'll just take that entire file and disable in Lazy itself,

**Reply by dpetka2001** [reply, marked as answer]
I just created a file in `~/.config/nvim/lua/plugins/go.lua
```lua
return {
  {
    "PerMalmberg/nvim-dap-go",
    branch = "add-lint-ci",
  },
}
```
and then opened Lazy UI and pressed `C` (for check) and `U` (for update) and it got installed correctly. 
![2024-05-19_16-38](https://github.com/LazyVim/LazyVim/assets/12776461/c6d20f0f-c69c-4234-bc41-7c9224e011fd)

Choose whatever branch you'd like.

PS: the above test was made with LazyExtra for Go enabled.

**Reply by PerMalmberg** [reply]
Ah, that makes so much sense. Custom files override the default ones. Awesome, thank you!
